private/bufpkg/bufplugin/bufpluginconfig/bufpluginconfig.go

```diff
@@ -192,6 +193,36 @@ type MavenRuntimeConfig struct {
 	Options []string
 }
 
+// SwiftRegistryConfig is the registry configuration for a Swift plugin.
+type SwiftRegistryConfig struct {
+	// Dependencies are dependencies for the remote package.
+	Dependencies []SwiftRegistryDependencyConfig
+}
+
+// SwiftRegistryDependencyConfig is the swift registry dependency configuration.
+type SwiftRegistryDependencyConfig struct {
+	// Package is the name of the Swift package.
+	Package string
+	// Version is the version of the Swift package.
+	Version string
+	// Products are the names of the products available to import.
+	Products []string
+	// Platforms are the minimum versions for platforms the package supports.
+	Platforms SwiftRegistryDependencyPlatformConfig
+}
+
+// SwiftRegistryDependencyPlatformConfig is the swift registry dependency platform configuration.
+type SwiftRegistryDependencyPlatformConfig struct {
+	// macOS specifies the version of the macOS platform.
+	MacOS string
+	// iOS specifies the version of the iOS platform.
+	IOS string
+	// TVOS specifies the version of the tvOS platform.
+	TVOS string
+	// WatchOS specifies the version of the watchOS platform.
+	WatchOS string
```

Comment:
How stable is this list? Ideally we wouldn't need to update our CLI + schema + other systems to support new platform types.

Reply:
@pkwarren as stable as apple's list of product offerings I suppose. I suppose this has only really been updated once this decade with the launch of watchOS.